oracle/oracle.go

```diff
@@ -444,3 +458,125 @@ func (o *oracleOps) Delete(volumeID string) error {
 	}
 	return nil
 }
+
+func (o *oracleOps) SetInstanceGroupSize(instanceGroupID string, count int64, timeout time.Duration) error {
+
+	ctx := context.Background()
+	var cancel context.CancelFunc
+	if timeout > time.Nanosecond {
+		// If timeout is non-zero then
+		// add deadline to the operation else
+		// wait infinitely for operation to complete
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+		defer cancel()
+	}
+
+	instanceGroupSize := int(count)
+
+	//get containerengine client
+	client, err := containerengine.NewContainerEngineClientWithConfigurationProvider(common.DefaultConfigProvider())
+	if err != nil {
+		return err
+	}
+
+	//get nodepool by ID to be updated
+	nodePoolReq := containerengine.ListNodePoolsRequest{CompartmentId: &o.compartmentID, Name: &instanceGroupID, ClusterId: &o.clusterID}
+	nodePools, err := client.ListNodePools(context.Background(), nodePoolReq)
+	if err != nil {
+		return err
+	}
+
+	if len(nodePools.Items) == 0 {
+		return errors.New("No node pool found with name " + instanceGroupID)
+	}
+	numberOfDomains := len(nodePools.Items[0].NodeConfigDetails.PlacementConfigs)
+	totalClusterSize := numberOfDomains * instanceGroupSize
+	logrus.Println("Setting instanceGroupSize to ", totalClusterSize, " in total ", numberOfDomains, " regions.")
+
+	//get all availabliity domain
+	nodePoolPlacementConfigDetails := make([]containerengine.NodePoolPlacementConfigDetails, numberOfDomains)
+
+	for i, placementConfigs := range nodePools.Items[0].NodeConfigDetails.PlacementConfigs {
+		nodePoolPlacementConfigDetails[i].AvailabilityDomain = placementConfigs.AvailabilityDomain
+		nodePoolPlacementConfigDetails[i].SubnetId = placementConfigs.SubnetId
+	}
+
+	//update node pools
+	req := containerengine.UpdateNodePoolRequest{
+		NodePoolId: nodePools.Items[0].Id, //get node pool id
+		UpdateNodePoolDetails: containerengine.UpdateNodePoolDetails{
+			NodeConfigDetails: &containerengine.UpdateNodePoolNodeConfigDetails{
+				Size:             &totalClusterSize,
+				PlacementConfigs: nodePoolPlacementConfigDetails,
+			},
+		},
+	}
+
+	resp, err := client.UpdateNodePool(context.Background(), req)
+	if err != nil {
+		return err
+	}
+
+	err = o.waitNodeStatus(resp.OpcRequestId, resp.OpcWorkRequestId, client)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *oracleOps) waitNodeStatus(opcRequestID, opcWorkRequestID *string, client containerengine.ContainerEngineClient) error {
+	workReq := containerengine.GetWorkRequestRequest{OpcRequestId: opcRequestID,
+		WorkRequestId: opcWorkRequestID}
+
+	f := func() (interface{}, bool, error) {
+		workResp, err := client.GetWorkRequest(context.Background(), workReq)
```

Comment:
[GetNodePool](https://docs.oracle.com/en-us/iaas/api/#/en/containerengine/20180222/NodePool/GetNodePool) API needs to be used to check lifecycle state of node pool is `ACTIVE`.

As additional check, (if required,) call `GetInstanceGroupSize` to verify number of active worker nodes matches with expected count.

Reply:
The state of work request changes from Accepted-> InProgress -> Succeeded, till all the instances are active. Oracle SDK provides this functionality to track the status  of any work request submitted till done.